mesonbuild/compilers/mixins/clike.py

```diff
@@ -697,7 +697,7 @@ def get_define(self, dname: str, prefix: str, env: 'Environment',
         star_idx = p.stdout.find(delim_start)
         end_idx = p.stdout.rfind(delim_end)
         if (star_idx == -1) or (end_idx == -1) or (star_idx == end_idx):
-            raise AssertionError('BUG: Delimiters not found in preprocessor output!')
+            raise MesonBugException(f'Delimiters ({delim_start}/{delim_end}) not found in preprocessor output.')
```

Comment:
```suggestion
            raise mesonlib.MesonBugException(f'Delimiters ({delim_start}/{delim_end}) not found in preprocessor output.')
```

Reply:
Done, and removed the delimiters as they contain newlines which looks confusing in the error message.